docs/api/implementation-guide.md

```diff
@@ -0,0 +1,142 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+In some cases, you may want to implement your own NuGet package feed.
+Many [existing implementations exist](../hosting-packages/Overview.md) which allow you to host your own feed in a variety way, but the protocol between the official NuGet client software and a package feed is documented allowing you to build your own feed implementation from scratch.
+
+The protocol does evolve over time and this guide is aimed at those that wish to or already have implemented a NuGet package server.
+
+Since the initial release of the NuGet V3 protocol in 2015, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value their package consumers, beyond simply exacting metadata from hosted packages and returning the metadata in various forms.
+For example, the search and package metadata endpoints contain more than just metadata found in the nupkg's nuspec file.
+
+Note that this guide is focused on the NuGet V3 protocol since the V2 protocol is essentially undocumented and, since 2015, the recommended protocol for NuGet client and server communication is the V3 protocol. For more information read about protocol versioning.
```

Comment:
This is fine but could be more authoritative. 

i.e. This guide is for NuGet V3 protocol. NuGet v2 protocol is [deprecated](https://learn.microsoft.com/en-us/nuget/nuget-org/overview-nuget-org#api-endpoint-for-nugetorg).

Reply:
We'll update the main branch in another PR.